plugins/modules/zendesk_ticket.py

```diff
@@ -0,0 +1,262 @@
+                'msg': to_native(e)
+            }
+
+    def close_ticket(self, ticket_id, status, body):
+        """
+        Closes or resolves a ticket in Zendesk.
+
+        Args:
+            ticket_id (int): The ID of the ticket to be closed or resolved.
+            status (str): The new status for the ticket ('closed' or 'resolved').
+            body (str): An optional comment to add to the ticket.
+
+        Returns:
+            dict: A dictionary containing the result of the ticket update operation.
+        """
+        url = '{0}/api/v2/tickets/{1}'.format(self.host, ticket_id)
+        payload = {
+            "ticket": {
+                "status": status,
+                "comment": {
+                    "body": body
+                }
+            }
+        }
+
+        request = self.api_aut()
+
+        try:
+            response = request.patch(url, data=json.dumps(payload))
+            if response.getcode() in [200, 204]:
+                return {
+                    'changed': True,
+                    'response': json.load(response),
+                }
+        except Exception as e:
+            return {
+                'changed': False,
+                'msg': to_native(e)
+            }
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(type='str', required=True),
+            username=dict(type='str', required=True, aliases=['user']),
+            password=dict(type='str', required=False, aliases=['pass'], no_log=True),
+            status=dict(type='str', required=True, choices=['new', 'closed', 'resolved']),
+            body=dict(type='str', default=''),
+            priority=dict(type='str', choices=['urgent', 'high', 'normal', 'low'], default='normal'),
+            subject=dict(type='str'),
+            token=dict(type='str', required=False, no_log=True),
+            ticket_id=dict(type='int')
+        ),
+        supports_check_mode=False
+    )
+
+    host = module.params['host']
+    username = module.params['username']
+    password = module.params['password']
+    status = module.params['status']
+    body = module.params['body']
+    priority = module.params['priority']
+    subject = module.params['subject']
+    token = module.params['token']
+    ticket_id = module.params['ticket_id']
+
+    if not password and not token:
+        module.fail_json(msg="Either 'password' or 'token' must be provided.")
+
+    zendesk_api = ZENDESK_API(username, password, token, host)
+
+    if status == 'new':
+        if not subject or not priority:
+            module.fail_json(msg="Both 'subject' and 'priority' must be provided when creating a new ticket.")
+        result = zendesk_api.create_ticket(body, priority, subject)
+    elif status in ['closed', 'resolved']:
+        if not ticket_id:
+            module.fail_json(msg="The 'ticket_id' must be provided when the status is 'closed'")
+        result = zendesk_api.close_ticket(ticket_id, status, body)
```

Comment:
I don't see any idempotence checks here, like "is the ticket already closed" before closing it, or "does the ticket already exist" before creating it.

Reply:
@felixfontein comment still stands - it is not resolved. I believe it could be solved somewhat easily by adding the state as a parameter to `check_ticket` and verifying whether the ticket state is different.